Add new RiotID routes; Capture RiotIDGameName from MatchV5

harmonherring opened the pull request:
Since it seems #59 is dead I've opened this PR. It's pretty much identical besides a few minor fixes

- Fixes a few problems that kept new endpoints from working
- Fixes log fields named incorrectly
- Fixes `by-riot-id` endpoint path
- Captures RiotIDGameName from MatchV5 endpoint

harmonherring commented on riot/account/account.go:
```diff
@@ -0,0 +1,53 @@
+)
+
+// AccountClient provides methods for the account endpoints of the League of Legends API.
+type AccountClient struct { //nolint:golint
```
@KnutZuidema in the other PR you mentioned this should be renamed to `Client`. However, `account.Client` already exists in [account/client.go](https://github.com/KnutZuidema/golio/pull/60/files#diff-e7bdc4af264de5c189e54d0eaf81796f1bf0980087ade9ac9afb9fb4fe93d484R6-R8) so that this package conforms to the same structure in the `lol` package.

I can change this if you want, but if another `account`-family endpoint(s) is added by riot at some point it may be useful to keep this structure? Up to you

KnutZuidema replied:
Fair point. I would assume that the endpoint hierarchy for accounts will stay flat, so let's just implement all the actions directly on the `Client` type. If another set of endpoints is added later, we'll have to refactor.

harmonherring-pro replied:
[1bce0c4](https://github.com/KnutZuidema/golio/pull/60/commits/1bce0c4b7da4f918608bf228137d61631cf35cea)

harmonherring replied:
[1bce0c4](https://github.com/KnutZuidema/golio/pull/60/commits/1bce0c4b7da4f918608bf228137d61631cf35cea)

KnutZuidema commented:
@harmonherring, not sure how it got in there, but the [commit](https://github.com/KnutZuidema/golio/pull/60/commits/35c27d1e2235e347fbdab11989c8d0c73ffa3f96) `refactor: test response body` does not belong here. You can drop it from the branch.

Nico-Mayer commented:
Riot also stated in this [article](https://www.riotgames.com/en/DevRel/summoner-names-to-riot-id) that 

- /lol/summoner/v4/summoners/by-name/{summonerName}
- /tft/summoner/v1/summoners/by-name/{summonerName}

endpoints are deprecated from now on, so it may be a good idea to include a comment on this.

Nico-Mayer commented:
@KnutZuidema are there plans to merge this MR in the near future?

harmonherring-pro commented:
@KnutZuidema apologies for the delay on this 😅 I've implemented the account methods directly on `Client` as you asked and also rebased that odd commit out of history

harmonherring-pro commented:
> Riot also stated in this [article](https://www.riotgames.com/en/DevRel/summoner-names-to-riot-id) that
> 
> * /lol/summoner/v4/summoners/by-name/{summonerName}
> * /tft/summoner/v1/summoners/by-name/{summonerName}
> 
> endpoints are deprecated from now on, so it may be a good idea to include a comment on this.

I agree, let's do this in a separate PR though 😄

harmonherring commented:
@KnutZuidema apologies for the delay on this 😅 I've implemented the account methods directly on Client as you asked and also rebased that odd commit out of history

harmonherring commented:
> Riot also stated in this [article](https://www.riotgames.com/en/DevRel/summoner-names-to-riot-id) that
> 
> * /lol/summoner/v4/summoners/by-name/{summonerName}
> * /tft/summoner/v1/summoners/by-name/{summonerName}
> 
> endpoints are deprecated from now on, so it may be a good idea to include a comment on this.

I agree, let's do this in a separate PR 😄

KnutZuidema reviewed (approved):
Thanks @harmonherring 🙏🏼 LGTM
